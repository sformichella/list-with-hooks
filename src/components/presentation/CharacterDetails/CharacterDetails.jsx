import React from 'react';
import PropTypes from 'prop-types';

export default function Character({ name, image, status }) {
  return (
    <div data-testid="character">
      <h1>{name}</h1>
      <img src={image}/>
      <p>{status}</p>
    </div>
  );
}

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired
};
