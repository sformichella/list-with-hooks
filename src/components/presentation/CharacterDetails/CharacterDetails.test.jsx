import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetails from './CharacterDetails';

describe('CharacterDetails component', () => {
  afterEach(() => cleanup());
  it('renders CharacterDetails', () => {
    const { asFragment } = render(<CharacterDetails />);
    expect(asFragment()).toMatchSnapshot();
  });
});
