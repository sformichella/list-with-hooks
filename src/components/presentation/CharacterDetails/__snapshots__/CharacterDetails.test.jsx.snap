// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetails component renders CharacterDetails 1`] = `
<DocumentFragment>
  <div
    data-testid="character"
  >
    <h1>
      Cool Name
    </h1>
    <img
      src="image.com"
    />
    <p>
      Alive
    </p>
  </div>
</DocumentFragment>
`;
